### PR TITLE
feat(kit): `Carousel` add new component

### LINKED ISSUE #257: [FEATURE] Carousel
# 🚀 Feature request

### Is your feature request related to a problem?

We need to add a carousel component

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring
-   [ ] Code style update
-   [ ] Build or CI related changes
-   [ ] Documentation content changes

## What is the current behavior?

Closes #257

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/kit/components/carousel/carousel-draggable.directive.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @HostBinding('class._draggable')
+    @tuiDefaultProp()
+    draggable = true;
+
+    @HostBinding('class._transitioned')
+    transitioned = false;
+
+    constructor(
+        @Inject(DOCUMENT) private readonly documentRef: Document,
+        @Inject(TUI_IS_MOBILE) private readonly isMobile: boolean,
```
Comment (splincode) [inline projects/kit/components/carousel/carousel-draggable.directive.ts:51]: do we also have to do like here?
https://github.com/TinkoffCreditSystems/taiga-ui/pull/1055/files?

File: projects/kit/components/carousel/carousel-scroller.directive.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    index = 0;
+
+    @Input()
+    current = 0;
+
+    constructor(
+        @Optional()
+        @Inject(TuiCarouselDraggableDirective)
+        readonly draggable: TuiCarouselDraggableDirective | null,
+        @Inject(TUI_IS_MOBILE) readonly isMobile: boolean,
```
Comment (splincode) [inline projects/kit/components/carousel/carousel-scroller.directive.ts:42]: do we also have to do like here?
https://github.com/TinkoffCreditSystems/taiga-ui/pull/1055/files?

**Review by splincode — APPROVED** [review]

File: projects/kit/components/carousel/carousel-scroller.directive.ts (hunk 2 above)
Comment (waterplea) [reply projects/kit/components/carousel/carousel-scroller.directive.ts:42]: No, this is not screen dependent but pointer dependent. It is unnecessary on touch devices.

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/components/carousel/carousel.template.html [hunk 3]
```diff
@@ -0,0 +1,129 @@
+<tui-doc-page header="Carousel" package="KIT">
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/carousel/carousel.template.html:1]:
```
type="components"
```
(link doesnt work properly)

File: projects/demo/src/modules/app/pages.ts [hunk 4]
```diff
@@ -206,6 +206,12 @@ export const pages: TuiDocPages = [
             'карта, карточка, card, visa, mastercard, credit, icon, logo, дебетовая, кредитная, иконка, логотип',
         route: '/components/card',
     },
+    {
+        section: $localize`Components`,
+        title: 'Carousel',
+        keywords: 'молекулы, карусель, слайдер, slider, carousel',
```
Comment (nsbarsukov) [inline projects/demo/src/modules/app/pages.ts:212]: why "молекулы" ?

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/app/pages.ts (hunk 4 above)
Comment (waterplea) [reply projects/demo/src/modules/app/pages.ts:212]: Old copy-paste :D

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**